src/main/resources/public/index.js

```diff
@@ -0,0 +1,59 @@
+const StoreList = {
+    data() {
+        return {
+            stores: [],
+            currentStore: null,
+            currentPage: 1,
+            nameSearch: null,
+            newStoreName: null,
+        }
+    },
+    watch: {
+        nameSearch() {
+            this.getStores()
+            this.currentPage = 1
+            this.currentStore = null
+        },
+        currentPage() {
+            this.currentStore = null
+            this.getStores()
+        },
+        currentStore(val) {
+            if (val && val.name === null) val.name = '';
+            this.newStoreName = null
+        },
+    },
+    methods: {
+        nextPage() {
+            this.currentPage++
+        },
+        previousPage() {
+            this.currentPage--
+        },
+        getStores() {
+            axios
+                .get('/stores', {
+                    params: {
+                        page: this.currentPage,
+                        nameSearch: this.nameSearch,
+                    }
+                })
+                .then(response => this.stores = response.data)
+        },
+        confirmStoreName(storeId) {
+            axios
+                .patch(`/stores/${storeId}`, {newName: this.newStoreName})
+                .then(() => {
+                    this.currentStore.name = this.newStoreName
+                    this.newStoreName = null
+                })
+                .then(this.getStores)
```

Comment:
is there a reason to do this `getStores`?

Reply:
it was a quick "hack" to have the store name updated on the table. But I could have just manually updated (if there were no errors). Maybe that would have been better to reduce unneed network calls.